### PR TITLE
Add utilities for comparing versions

### PR DESCRIPTION
**Pull request by spadgett**
Take 2. I had to revert the other change because of a problem with the previous semver library and bower.

/assign @benjaminapetersen

### CODE REVIEW

File: test/spec/services/versionsServiceSpec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  var VersionsService;
+  beforeEach(function() {
+    inject(function(_VersionsService_) {
+      VersionsService = _VersionsService_;
+    });
+  });
+
+  var versions = ["10.1", "9.1", "10.0", "9.0", "8.0", "10.1.1", "0.10", "latest", "old"];
+
```
Comment (benjaminapetersen) [inline test/spec/services/versionsServiceSpec.js]: +1 tests, woot!

File: dist/origin-web-common.js [hunk 2]
```diff
@@ -5787,6 +5787,34 @@ angular.module('openshiftCommonServices')
...
+        // the values is not a version string.
+        var result = window.compareVersions(v1, v2);
+        return newestFirst ? result * -1 : result;
+      } catch (e) {
+        // One of the values is not a version string. Fall back to string comparison.
+        // Numbers will be sorted higher by localeCompare.
+        return v1.localeCompare(v2);
+      }
+    };
+
```
Comment (benjaminapetersen) [inline dist/origin-web-common.js]: Quick Q ~ What do we get out of `rcompare`, assuming it means `reverseCompare`, correct?

Comment (spadgett) [reply dist/origin-web-common.js]: Yeah, the intention is that you can pass it to Arrays.sort. If you have to pass a 3rd param, that doesn't work.

Let me refactor this a bit to make it so compare and rcompare both take just the v1 and v2 params.

**Comment by spadgett** [comment]
@benjaminapetersen updated

File: dist/origin-web-common.js (hunk 2 above)
Comment (benjaminapetersen) [reply dist/origin-web-common.js]: Nice, this looks good to me!

**Comment by benjaminapetersen** [comment]
@spadgett lgtm, thanks!

**Pull request merged**